web/pdf_thumbnail_view.js

```diff
@@ -191,7 +190,16 @@ var PDFThumbnailView = (function PDFThumbnailViewClosure() {
       this.div.setAttribute('data-loaded', true);
       this.ring.appendChild(canvas);
 
-      return canvas.getContext('2d');
+      var ctx = canvas.getContext('2d');
+      var outputScale = getOutputScale(ctx);
+      canvas.width = this.canvasWidth * outputScale.sx;
```

Comment:
Please make sure that `canvas.width` is an integer, by changing this line to:
```javaScript
canvas.width = (this.canvasWidth * outputScale.sx) | 0;
```
And similar for `canvas.height` on the next line.